src/Tribe/Editor/Meta.php

```diff
@@ -14,6 +14,11 @@ class Tribe__Events__Editor__Meta extends Tribe__Editor__Meta {
 	 * @return void
 	 */
 	public function register() {
+
+		// Provide backwards compatibility for meta data
+		$post_type = Tribe__Events__Main::POSTTYPE;
+		add_filter( "rest_prepare_{$post_type}", array( $this, 'meta_backwards_compatibility' ), 10, 3 );
```

Comment:
Thanks @mitogh !

No, the problem here is that the new app is effectively using boolean, but the data we inherit from old sites who have events with that info already in their databases is "boolean" but using strings. Boolean need to keep boolean and this should be treated as a particular case. 

If we define a new type, we will still have to convert the values that we already have in the database in some way. And actually, after testing millions of different ways, this seems to be the more optimal. The sanitization doesn't apply when getting.

Reply:
OKay makes sense thanks for clarifying I think only thing I can flag is to check if `$data->data['meta']['_EventAllDay']` has been set using either `isset` or `empty`

Reply:
Great question, it's basically as a self defense mechanism in case `_EventAllDay` is removed from the `$data['meta']` value by external filter and by using `isset` we ensure the value is there before accessing and same for `meta` I think using `isset( $data['meta'], $data['meta']['_EventAllDay']` i was mentioned `empty` as well as works same as `issset` not suggesting that the data was necessarily empty but rather not defined. 

Let me know if you have additional questions on that, thanks.

Reply:
Thanks @mitogh !

Yes, I get it, but as I mentioned before, if there's no value, then we don't get into the `if`. We already check if there's metadata defined for that post on that if -> https://github.com/moderntribe/the-events-calendar/pull/2283/files#diff-be71bbdff8ba1082c8ba7f700139f060R112

If there's value, then we are absolutely sure that the data is there.

You mean if somebody actually adds a filter to alter the REST response and remove that index before we do it? Then they can change the priority of the filter and remove it anyway.

Reply:
Oh! I see what you mean however even if the post has the meta but if the field is not registered like: 

- `register_meta( 'post', '_EventAllDay', $this->boolean() );` the value is not going to be present on `$data` but I think as we are doing that explicitly into the plugin if someone removes that from the response (which seems unlikely and I don't see why that might be the case when the error might be present) so I don't think we should consider that scenario as "normal".

Thanks for the clarification.